modules/geo-location/class.jetpack-geo-location.php

```diff
@@ -0,0 +1,432 @@
+			return '';
+		}
+
+		$html = '<div class="post-geo-location-label geo-chip">';
+		$html .= '<span class="dashicons dashicons-location" style="vertical-align: text-top;"></span> ';
+		$html .= esc_html( $meta_values['label'] );
+		$html .= '</div>';
+
+		/**
+		 * Allow modification or replacement of the default geo-location display HTML.
+		 *
+		 * @module geo-location
+		 *
+		 * @param array $html The default HTML for displaying a geo-location label.
+		 * @param array $geo_data An array containing "latitude", "longitude" and "label".
+		 */
+		$html = apply_filters( 'jetpack_geo_location_display', $html, $meta_values );
+
+		return $html;
+	}
+
+	/**
+	 * Get the ID of the current global post object, if available.  Otherwise, return null.
+	 *
+	 * This isolates the access of the global scope to this single method, making it easier to
+	 * safeguard against unexpected missing $post objects in other hook functions.
+	 *
+	 * @return int|null
+	 */
+	public function get_post_id() {
+		global $post;
+
+		if ( ! isset( $post) || ! $post || ! is_object( $post ) || ! isset( $post->ID ) ) {
+			return null;
+		}
+
+		return $post->ID;
+	}
+
+	/**
+	 * This method always returns an array with the following structure:
+	 *
+	 * array(is_public => bool, latitude => float, longitude => float, label => string, is_populated => bool)
+	 *
+	 * So, regardless of whether your post actually has values in postmeta for the geo-location fields,
+	 * you can be sure that you can reference those array keys in calling code without having to juggle
+	 * isset(), array_key_exists(), etc.
+	 *
+	 * Mocking this method during testing can also be useful for testing output and logic in various
+	 * hook functions.
+	 *
+	 * @param integer $post_id
+	 *
+	 * @return array A predictably structured array representing the meta values for the supplied post ID.
+	 */
+	public function get_meta_values( $post_id ) {
+		$meta_values = array(
+			'is_public'    => (bool) $this->sanitize_public( $this->get_meta_value( $post_id, 'public' ) ),
+			'latitude'     => $this->sanitize_coordinate( $this->get_meta_value( $post_id, 'latitude' ) ),
+			'longitude'    => $this->sanitize_coordinate( $this->get_meta_value( $post_id, 'longitude' ) ),
+			'label'        => trim( $this->get_meta_value( $post_id, 'address' ) ),
+			'is_populated' => false,
+		);
+
+		if ( $meta_values['latitude'] && $meta_values['longitude'] && $meta_values['label'] ) {
+			$meta_values['is_populated'] = true;
+		}
+
+		return $meta_values;
+	}
+
+	/**
+	 * This function wraps get_post_meta() to enable us to keep the "geo_" prefix isolated to a single
+	 * location in the code and to assist in mocking during testing.
+	 *
+	 * @param integer $post_id
+	 * @param string $meta_field_name
+	 *
+	 * @return mixed
+	 */
+	public function get_meta_value( $post_id, $meta_field_name ) {
+		if ( ! $post_id ) {
+			return null;
+		}
+
+		return get_post_meta( $post_id, 'geo_' . $meta_field_name, true );
+	}
+
+	/**
```

Comment:
There seem to be a few `Simple wrapper for testing purposes.` functions here. Ideally we'd have nothing in our implementation that's solely there for the unit tests.

What do you think? Maybe we can get rid of them and mock these?

Reply:
007f253 takes a new approach to mocking these core functions that keeps all the test-related code in the unit test case rather than the main implementation.

Reply:
Thanks! This is a neat approach.